src/main/editDelete/preProcess.ts

```diff
@@ -102,6 +103,21 @@ const addSpaces: (totalDuration: number) => MapCallback<Word, Word[]> =
     return wordAndSilence;
   };
 
+const calculateAverageSilenceDuration = (
+  jsonTranscription: JSONTranscription
+): number => {
+  let silenceSum = 0;
+  for (let i = 0; i < jsonTranscription.words.length - 1; i += 1) {
+    const endTime = jsonTranscription.words[i + 1].start_time;
+    const silenceDuration =
+      endTime -
+      jsonTranscription.words[i].start_time -
+      jsonTranscription.words[i].duration;
+    silenceSum += silenceDuration;
+  }
+  return silenceSum / jsonTranscription.words.length;
```

Comment:
If the transcript is empty does this break the app from a div0?

Reply:
Good pick up. I've replaced this with:
  return jsonTranscription.words.length !== 0
    ? silenceSum / jsonTranscription.words.length
    : totalDuration;
as the average silence duration just equals the total duration if the transcript has no words